xt/48.1-ledgersmb-report-unapproved-batch_overview.t

```diff
@@ -0,0 +1,137 @@
+
+    if($batch_data->{__POST}) {
+        $batch = LedgerSMB::Batch->new({ base => {
+            dbh => $dbh,
+            batch_id => $batch_id,
+        }});
+        $batch->post or BAIL_OUT 'Failed to post/approve test batch';
+    }
+}
+
+
+plan tests => (36);
+
+
+# Initialise Object
+$report = LedgerSMB::Report::Unapproved::Batch_Overview->new();
+isa_ok($report, 'LedgerSMB::Report::Unapproved::Batch_Overview', 'instantiated object');
+ok($report->set_dbh($dbh), 'set dbh');
+
+# Query with no filter
+ok($rows = $report->get_rows, 'get_rows() called ok');
+is(ref $rows, 'ARRAY', 'get_rows() returns an arrayref');
+is($rows, $report->rows, 'rows property is set by run_report()');
+is(ref $report->rows, 'ARRAY', 'rows property is an arrayref');
+is(scalar @{$report->rows}, scalar @test_batches, 'returned all rows');
+
+# Query description - should return just 1 row
+ok($report->description('AP batch'), 'set description');
+ok($rows = $report->get_rows, 'get_rows() with description');
+is(scalar @{$rows}, 1, 'querying description returned 1 row');
+$row = $$rows[0];
+is(ref $row, 'HASH', 'first returned element is a hashref');
+ok($row->{id}, 'row id field is defined');
+is($row->{id}, $row->{row_id}, 'row_id field matches id field');
+is($row->{batch_class}, 'ap', 'row batch_class field is correct');
+is($row->{control_code}, 'TEST-001', 'row control_code field is correct');
+is($row->{description}, 'Test AP batch Description', 'row payment field is correct');
+is($row->{default_date}, '2018-09-08', 'row default_date field is correct');
+is($row->{transaction_total}, 0, 'row transaction_total field is correct');
+is($row->{payment_total}, 0, 'row payment_total field is correct');
+like($row->{created_on}, qr/\d{4}-\d{2}-\d{2}/, 'row created_on field is of expected format');
+ok(exists $row->{created_by}, 'row created_by field exists');
+ok(exists $row->{lock_success}, 'row lock_success field exists');
+is($report->description(undef), undef, 'reset description filter');
+
+# Query class_id
+ok($report->class_id(3), 'set class_id filter');
+ok($rows = $report->get_rows, 'get_rows() with batch_class');
+is(scalar @{$rows}, 1, 'querying with class_ids returned 1 row');
+$row = $$rows[0];
+is($row->{description}, 'Test Payment batch description', 'row control_code field is correct');
+is($report->class_id(undef), undef, 'reset class_id filter');
+
+# Query approved
+is($report->approved(1), 1, 'set approved filter = true');
+ok($rows = $report->get_rows, 'get_rows() with approved_filter = true');
+is(scalar @{$rows}, 1, 'querying with approved=true returned 1 row');
+is($report->approved(undef), undef, 'reset approved filter');
+
+# Query unapproved
+is($report->approved(0), 0, 'set approved filter = false');
+ok($rows = $report->get_rows, 'get_rows() with approved_filter = false');
+is(scalar @{$rows}, 3, 'querying with approved=false returned 3 rows');
+is($report->approved(undef), undef, 'reset approved filter');
+
+
+#TODO Query Amount
```

Comment:
Do we need to create an issue for this in GitHub? If we don't, how are we going to track what items are remaining?

Or is there work in progress and should I hold back on merging?

Reply:
I would say no need to raise an issue in GitHub. We find the issue again by grepping `TODO` in our source code - and there's plenty like it. This isn't a bug or user-affecting. Our current unit test coverage is so patchy, I think it makes no sense to raise GitHub issues for missing unit test coverage until we're nearer 100%.

There's no further work in progress on this test. If you prefer, I could add the missing part, but that makes it more difficult for you in merging MC.